pkgs/applications/misc/genxword/default.nix

```diff
@@ -0,0 +1,48 @@
+{ lib
+, python3
+, fetchFromGitHub
+, gettext
+, gobject-introspection
+, wrapGAppsHook
+, pango
+, gtksourceview3
+}:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "genxword";
+  version = "2.0.1";
+
+  src = fetchFromGitHub {
+    owner = "riverrun";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "00czdvyb5wnrk3x0g529afisl8v4frfys9ih0nzf1fs4jkzjcijg";
+  };
+
+  nativeBuildInputs = [
+    gettext
+    gobject-introspection
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    gobject-introspection
+    pango
+    gtksourceview3
+  ];
+
+  propagatedBuildInputs = with python3.pkgs; [
+    pycairo
+    pygobject3
+  ];
```

Comment:
Currently, the executables are wrapped twice, so:

```diff
diff --git i/pkgs/applications/misc/genxword/default.nix w/pkgs/applications/misc/genxword/default.nix
index ff2ec537d61..215542003ad 100644
--- i/pkgs/applications/misc/genxword/default.nix
+++ w/pkgs/applications/misc/genxword/default.nix
@@ -36,6 +36,12 @@ python3.pkgs.buildPythonApplication rec {
     pygobject3
   ];
 
+  # to prevent double wrapping
+  dontWrapGApps = true;
+  preFixup = ''
+    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
+  '';
+
   # there are no tests
   doCheck = false;
 ```

Reply:
Nice trick! But is it really worth it?